cmd/dp-recipe-api/main.go

```diff
@@ -38,30 +53,119 @@ func main() {
 
 	log.Event(ctx, "config on startup", log.INFO, log.Data{"config": cfg})
 
+	versionInfo, healthErr := health.NewVersionInfo(
+		BuildTime,
+		GitCommit,
+		Version,
+	)
+	if healthErr != nil {
+		log.Event(ctx, "failed to create service version information", log.ERROR, log.Error(healthErr))
+		os.Exit(1)
+	}
+
+	hc := health.New(versionInfo, cfg.HealthCheckCriticalTimeout, cfg.HealthCheckInterval)
+
 	//Feature Flag for Mongo Connection
 	enableMongoData := cfg.MongoConfig.EnableMongoData
 
+	datastore := &store.DataStore{Backend: nil}
+
+	mongodb := &mongo.Mongo{
+		Collection: cfg.MongoConfig.Collection,
+		Database:   cfg.MongoConfig.Database,
+		URI:        cfg.MongoConfig.BindAddr,
+	}
+
+	var err error
 	if enableMongoData {
-		mongodb := &mongo.Mongo{
-			Collection: cfg.MongoConfig.Collection,
-			Database:   cfg.MongoConfig.Database,
-			URI:        cfg.MongoConfig.BindAddr,
-		}
 
-		var err error
 		mongodb.Session, err = mongodb.Init()
 		if err != nil {
 			log.Event(ctx, "failed to initialise mongo", log.FATAL, log.Error(err))
 			os.Exit(1)
+		} else {
+			log.Event(ctx, "listening to mongo db session", log.INFO, log.Data{
+				"bind_address": cfg.BindAddr,
```

Comment:
The bind address is not related to the mongo DB connection? I don't get why its part of the log data here.

Reply:
Rectified in the commit "Change bind address when mongo connected in log message to mongo bind address"